extensions/amp-analytics/0.1/iframe-transport.js

```diff
@@ -138,6 +145,40 @@ export class IframeTransport {
     return frameData;
   }
 
+  createLongTaskObserver_() {
+    if (this.longTaskObserver_ || typeof PerformanceObserver == 'undefined') {
+      return;
+    }
+    // TODO(jonkeller): Re-enable linter after
+    // https://github.com/sindresorhus/globals/pull/120 merges
+    // eslint-disable-next-line no-undef
+    this.longTaskObserver_ = new PerformanceObserver(entryList => {
+      if (!entryList) {
+        return;
+      }
+      entryList.getEntries().forEach(entry => {
+        if (entry && entry['entryType'] == 'longtask') {
+          if (entry['name'] == 'cross-origin-descendant' ||
+              entry['name'] == 'multiple-contexts' ||
+              (getMode().localDev &&
+              entry['name'] == 'same-origin-descendant')) {
+            user().warn(TAG_, `${this.type_} frame took ${entry.duration}ms`);
+            if (entry.attribution) {
+              entry.attribution.forEach(attrib => {
```

Comment:
TODO: Test with multiple fake vendors, each with an artificial delay, and see what happens here.